src/QMCHamiltonians/StaticStructureFactor.cpp

```diff
@@ -106,7 +106,7 @@ void StaticStructureFactor::addObservables(PropertySetType& plist, BufferType& c
 
 void StaticStructureFactor::registerCollectables(std::vector<ObservableHelper>& h5desc, hdf_archive& file) const
 {
-  h5desc.emplace_back(std::vector<std::string>{name_, "kpoints"});
+  h5desc.push_back({{name_, "kpoints"}});
```

Comment:
Is there a reason for when to use a plain character string vs. when to use the string literal 's' operator? (as in SkAllEstimator.cpp)

Reply:
Whether its required or not is a question of when the complier can successfully infer it or not.

[Cpp core guidelines has a bit to say]
(http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#slstr12-use-the-s-suffix-for-string-literals-meant-to-be-standard-library-strings)

Which makes we think we should use it here and consistently which was an element of #4332.

There are a bunch of type resolution implications as well.  These are generally invisible since std::string has  a const char * constructor.